stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/channel/list/adapter/ChannelListItemAdapter.kt

```diff
@@ -0,0 +1,46 @@
+package io.getstream.chat.android.ui.channel.list.adapter
+
+import android.view.ViewGroup
+import io.getstream.chat.android.ui.R
+import io.getstream.chat.android.ui.channel.list.ChannelListViewStyle
+import io.getstream.chat.android.ui.channel.list.adapter.diff.ChannelItemDiff
+import io.getstream.chat.android.ui.channel.list.adapter.viewholder.BaseChannelListItemViewHolder
+import io.getstream.chat.android.ui.channel.list.adapter.viewholder.BaseChannelViewHolderFactory
+import io.getstream.chat.android.ui.utils.extensions.cast
+import io.getstream.chat.android.ui.utils.extensions.firstOrDefault
+
+public class ChannelListItemAdapter : BaseChannelListItemAdapter() {
+
+    public companion object {
+        public val DEFAULT_DIFF: ChannelItemDiff = ChannelItemDiff()
+    }
+
+    public var viewHolderFactory: BaseChannelViewHolderFactory<out BaseChannelListItemViewHolder>? = null
+
+    public override var style: ChannelListViewStyle? = null
+
+    private fun getChannelItemLayout(): Int =
+        style?.channelPreviewLayout ?: R.layout.stream_channel_list_item_view
+
+    override fun getItemCount(): Int = channels.count()
+
+    override fun getItemViewType(position: Int): Int =
+        viewHolderFactory?.getItemViewType(channels[position])
+            ?: throw IllegalStateException("Please provide a view holder factory")
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): BaseChannelListItemViewHolder =
+        viewHolderFactory?.createChannelViewHolder(getChannelItemLayout(), parent, viewType)?.also { vh ->
+            vh.style = style
+            vh.userClickListener = userClickListener
+            vh.channelLongClickListener = channelLongClickListener
+            vh.channelClickListener = channelClickListener
```

Comment:
Why don't we have all those attributes in the Factory?

Reply:
I agree. Wasn't sure how much I would be able to change things from the current API.

Reply:
I agreed initially, but I've since found some issues when trying to do it that way (https://getstream.slack.com/archives/G016YAWD423/p1605638235150100)

Reply:
For now we can leave it as it is, but we need to think about that. It doesn't work fine when different VHs live on the same adapter